Application/Education/ClassRegister/Diary/Frontend.php

```diff
@@ -0,0 +1,859 @@
+        } elseif ($tblGroup) {
+            if (($tblYearList = Term::useService()->getYearByNow())) {
+                $tblYear = reset($tblYearList);
+            } else {
+                $tblYear = false;
+            }
+
+            // Gruppeneinträge
+            if ($tblYear && ($tblDiaryList = Diary::useService()->getDiaryAllByGroup($tblGroup, $tblYear))) {
+                foreach ($tblDiaryList as $tblDiary) {
+                    $diaryList[$tblDiary->getId()] = $tblDiary;
+                }
+            }
+            $tblPersonList = Group::useService()->getPersonAllByGroup($tblGroup);
+        } else {
+            $tblPersonList = false;
+            $tblYear = false;
+        }
+
+        // zusätzliche Schülereintrage (z.B. vom Klassenwechsel)
+        if ($tblPersonList && $tblYear) {
+            foreach ($tblPersonList as $tblPerson) {
+                if (($tblDiaryListByStudent = Diary::useService()->getDiaryAllByStudent($tblPerson, $tblYear))) {
+                    foreach ($tblDiaryListByStudent as $item) {
+                        if (!isset($diaryList[$item->getId()])) {
+                            $diaryList[$item->getId()] = $item;
+                        }
+                    }
+                }
+            }
+        }
+
+        // sortieren nach Datum
+        $diaryList = $this->getSorter($diaryList)->sortObjectBy('Date', new DateTimeSorter(), Sorter::ORDER_DESC);
+        $count = 0;
+        /** @var TblDiary $tblDiaryItem */
+        foreach ($diaryList as $tblDiaryItem) {
+            if ($tblStudentPerson) {
+                if (!Diary::useService()->getDiaryStudentAllByDiary($tblDiaryItem)
+                    || Diary::useService()->existsDiaryStudent($tblDiaryItem, $tblStudentPerson)
+                ) {
+                    // nur Klasseneinträge und welche mit dem ausgewählten Schüler
+                    $count++;
+                    $dataList[] = $this->setDiaryItem($tblDiaryItem, $count, $tblStudentPerson);
+                }
+            } else {
+                $count++;
+                $dataList[] = $this->setDiaryItem($tblDiaryItem, $count);
+            }
+        }
+
+        $columns = array(
+            'Number' => '#',
+            'Information' => 'Information',
+            'PersonList' => 'Schüler',
+            'Content' => 'Inhalt'
+        );
+
+        if (!$tblStudentPerson) {
+            $columns['Options'] = ' ';
+        }
+        return new TableData(
+            $dataList,
+            null,
+            $columns,
+            array(
+                'order' => array(
+                    array(0, 'asc')
+                ),
+                'responsive' => false
+            )
+        );
+    }
+
+    /**
+     * @param TblDiary $tblDiary
+     * @param int $count
+     * @param TblPerson|null $tblStudentPerson
+     *
+     * @return array
+     */
+    private function setDiaryItem(TblDiary $tblDiary, &$count, TblPerson $tblStudentPerson = null)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Incorrect spacing between argument "$tblStudentPerson" and equals sign; expected 0 but found 1](https://app.codacy.com/manual/gerdchristian-kunze/SPHERE-Framework/pullRequest?prid=5646354)